float8_experimental/float8_dynamic_linear.py

```diff
@@ -190,9 +191,20 @@ def __repr__(self):
         return f"WeightWithDynamicFloat8CastTensor(tensor={self._tensor}, mm_config={self._mm_config})"
 
     def fsdp_pre_all_gather(self, mesh):
```

Comment:
if `_pre_computed_amax`, we skip `tensor_to_amax` and directly do `amax_to_scale`